tests/starknet/executor.rs

```diff
@@ -0,0 +1,128 @@
+use std::{fs, thread};
+
+use anyhow::{anyhow, Error};
+use regex::Regex;
+
+use super::scarb::Compiler;
+
+#[allow(dead_code)]
+pub struct Executor {
+    accounts: Vec<String>,
+}
+
+#[allow(dead_code)]
+impl Executor {
+    pub fn new(num: usize) -> Result<Self, Error> {
+        let template = "./tests/starknet/contract/account".to_string();
+        let mut accounts = Vec::with_capacity(num);
+        accounts.push(template);
+        Ok(Self { accounts })
+    }
+
+    pub fn deploy_accounts(
+        &mut self,
+        update_template: bool,
+    ) -> Result<(), Error> {
+        if update_template {
+            self.update_account(&self.accounts[0])?;
+        }
+        self.prepare_contracts_environment()?;
+        self.compile()?;
+        // TODO
+        // #804 starkli signer keystore new key.json - Storing somewhere or deleting?
+        // #804 starkli account oz init account.json - Storing somewhere or deleting?
+        // #804 declare accounts
+        // #804 #805 fund accounts from pre-funded account
+        // #804 deploy accounts
+        // #806 iterate through class hashes and call getClass to see if they are verified
+        Ok(())
+    }
+
+    fn update_account(&self, path: &str) -> Result<(), Error> {
+        let lib_path = path.to_owned() + "/src/lib.cairo";
+        let account_old = fs::read_to_string(lib_path.clone())?;
+        let re = Regex::new(r"self.id.write\((?<number>\d+)\);")?;
```

Comment:
It might be easier to extract a constant in the `lib.cairo` and patch just it, not the actual code. Or add a placeholder instead of zero, as the contract is not expected to be runnable/deployable outside the executor. Then simple `replace` should be enough.

Reply:
I will go with the constant and not a placeholder, just for us so that we have stored information in the contract which was the last value used.
I wanted to have it as part of a contract so that maybe, some time in the future, we might have additional test cases that will involve interaction of this accounts that we have deployed, i.e. some marketplace.
Therefore, I will go with a constant, if it will be possible to call account and get the value of the constant. I was sure about this id, with creating function id(). I have to investigate whether this is the case for constants.

Reply:
@sergey-melnychuk Updated template contract, using placeholder <ID> and replacing it with timestamp that has format YMDHMS (e.g. 20241023140955).